docs/chugsplash-file.md

```diff
@@ -0,0 +1,62 @@
+# ChugSplash File
+
+A ChugSplash file is where you define an entire deployment or upgrade. It includes configuration settings, contract definitions, and state variables. The ChugSplash file replaces deployment scripts, which are used in other tools like [Hardhat](https://hardhat.org/hardhat-runner/docs/guides/deploying) and [Foundry](https://book.getfoundry.sh/tutorials/solidity-scripting).
+
+## Table of Contents
+
+- [Layout of a ChugSplash File](#layout-of-a-chugsplash-file)
+  - [Configuration settings](#configuration-settings)
+  - [Contract definitions](#contract-definitions)
+  - [State variables](#state-variables)
+
+## Layout of a ChugSplash File
+
+```ts
+{
+    // Configuration settings:
+    options: {
+        projectName: 'My ERC20 Project',
+    },
+    // Contract definitions:
+    contracts: {
+        // First contract definition:
+        MyToken: {
+            contract: 'ERC20', // Contract name in your Solidity file
+            variables: {
+                name: 'My Token',
+                symbol: 'MYT',
+                decimals: 18,
+                totalSupply: 1000,
+            }
+        },
+        // Second contract definition:
+        MyTokenRegistry: {
+            contract: 'ERC20Registry',
+            variables: {
+                tokenAddress: '{{ MyToken }}', // MyToken's address
+                ...
+            }
+        },
+        // Other contract definitions:
+        ...
+    }
+}
+```
+
+### Configuration settings
+
+The `options` property in your ChugSplash file contains only a single field, `projectName`. The project name can be any name you choose.
```

Comment:
Might want to specify that the project name needs to be unique. This is actually a limitation we should probably deal with b/c it will be quite annoying for users.

Reply:
We need project names to be unique because it determines the unique address of the ChugSplashManager (which in turn determines the unique addresses of all contracts deployed by it). I don't think this is too much of an issue. E.g. github repos have unique names

Reply:
Gonna add this to the techncial overview